src/SFML/Window/Unix/Display.cpp

```diff
@@ -42,27 +42,21 @@
 
 namespace
 {
-// The shared display and its reference counter
-Display*             sharedDisplay     = nullptr;
-unsigned int         referenceCount    = 0;
-XIM                  sharedXIM         = nullptr;
-unsigned int         referenceCountXIM = 0;
-std::recursive_mutex mutex;
-
-using AtomMap = std::unordered_map<std::string, Atom>;
-AtomMap atoms;
+std::shared_ptr<Display> sharedDisplay;
```

Comment:
The previous behavior would be better reproduced with an `std::weak_ptr`. For reference, we already use `std::weak_ptr` to manage shared OpenGL context in `GlContext.cpp`.

Reply:
My previous implementation was keeping the display and XIM alive forever because the global shared_ptr didn't go out of scope. Now I'm using a weak_ptr and reassigning that weak pointer when the underlying shared state gets rebuilt.